go-controller/pkg/cni/cni.go

```diff
@@ -138,16 +161,55 @@ func (pr *PodRequest) cmdAdd(kubeAuth *KubeAPIAuth, podLister corev1listers.PodL
 	return response, nil
 }
 
-func (pr *PodRequest) cmdDel() error {
-	if config.OvnKubeNode.Mode == types.NodeModeDPUHost {
-		// nothing to do
-		return nil
+func (pr *PodRequest) cmdDel(podLister corev1listers.PodLister, kclient kubernetes.Interface) (*Response, error) {
+	// assume success case, return an empty Result
+	response := &Response{}
+	response.Result = &current.Result{}
+
+	vfNetdevName := ""
+	if pr.CNIConf.DeviceID != "" {
+		if config.OvnKubeNode.Mode == types.NodeModeDPUHost {
+			pod, err := getPod(podLister, kclient, pr.PodNamespace, pr.PodName)
+			if err != nil {
+				klog.Errorf("Failed to get pod %s/%s: %v", pr.PodNamespace, pr.PodName, err)
+				return response, nil
+			}
+			dpuCD := util.DPUConnectionDetails{}
+			if err := dpuCD.FromPodAnnotation(pod.Annotations); err != nil {
+				klog.Errorf("Failed to get DPU connection details annotation for pod %s/%s: %v", pr.PodNamespace,
+					pr.PodName, err)
+				return response, nil
+			}
+			vfNetdevName = dpuCD.VfNetdevName
+		} else {
+			ovsIfName := pr.SandboxID[:15]
+			out, err := ovsGet("interface", ovsIfName, "external_ids", "vf-netdev-name")
+			if err != nil {
+				klog.Warningf("Couldn't find the original VF Netdev name from OVS interface %s for pod %s/%s: %v",
+					ovsIfName, pr.PodNamespace, pr.PodName, err)
+			} else {
+				vfNetdevName = out
+			}
+		}
 	}
 
-	if err := pr.PlatformSpecificCleanup(); err != nil {
-		return err
+	podInterfaceInfo := &PodInterfaceInfo{
+		IsDPUHostMode: config.OvnKubeNode.Mode == types.NodeModeDPUHost,
+		VfNetdevName:  vfNetdevName,
 	}
-	return nil
+	if !config.UnprivilegedMode {
```

Comment:
Yes, there is a difference. 

You cannot do any of the Opernations inside the network namespace when in Unprivileged mode. 

```
func (pr *PodRequest) UnconfigureInterface(ifInfo *PodInterfaceInfo) error {
	podDesc := fmt.Sprintf("for pod %s/%s", pr.PodNamespace, pr.PodName)
	klog.V(5).Infof("Tear down interface (%+v) %s", *pr, podDesc)
	if pr.CNIConf.DeviceID == "" {
		if ifInfo.IsDPUHostMode {
			klog.Warningf("Unexpected configuration %s, Device ID must be present for pod request on smart-nic host",
				podDesc)
			return nil
		}
	} else {
		// For SRIOV case, we'd need to move the VF from container namespace back to the host namespace
		netns, err := ns.GetNS(pr.Netns)
		if err != nil {
	
```
Those `netns` operations will not work.

Reply:
In cnishim.go, when `response.Result == nil` (which indicates UnprivilegedMode mode, since response.Result is set to nil in UnprivilegedMode mode), `UnconfigureInterface` is also called.

I mean `UnconfigureInterface` is called both in privileged mode and unprivileged mode, I don't see the netns operations are not executed in unprivileged mode, no?